File: BraintreeCore/src/test/java/com/braintreepayments/api/AnalyticsConfigurationUnitTest.java
package com.braintreepayments.api;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static junit.framework.Assert.assertEquals;
import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertNull;
import static junit.framework.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class AnalyticsConfigurationUnitTest {

    @Test
    public void fromJson_parsesFullInput() throws JSONException {
        JSONObject input = new JSONObject()
                .put("url", "https://example.com/analytics");

        AnalyticsConfiguration sut = AnalyticsConfiguration.fromJson(input);
        assertTrue(sut.isEnabled());
        assertEquals("https://example.com/analytics", sut.getUrl());
    }

    @Test
    public void fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        AnalyticsConfiguration sut = AnalyticsConfiguration.fromJson(null);
        assertFalse(sut.isEnabled());
        assertNull(sut.getUrl());
    }

    @Test
    public void fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        AnalyticsConfiguration sut = AnalyticsConfiguration.fromJson(new JSONObject());
        assertFalse(sut.isEnabled());
        assertNull(sut.getUrl());
    }
}

File: BraintreeCore/src/test/java/com/braintreepayments/api/AnalyticsConfigurationUnitTest.kt
package com.braintreepayments.api

import org.robolectric.RobolectricTestRunner
import org.json.JSONObject
import org.junit.Assert.*
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(RobolectricTestRunner::class)
class AnalyticsConfigurationUnitTest {
    @Test
    fun fromJson_parsesFullInput() {
        val input = JSONObject()
            .put("url", "https://example.com/analytics")
        val sut = AnalyticsConfiguration(input)
        assertTrue(sut.isEnabled)
        assertEquals("https://example.com/analytics", sut.url)
    }

    @Test
    fun fromJson_whenInputNull_returnsConfigWithDefaultValues() {
        val sut = AnalyticsConfiguration(null)
        assertFalse(sut.isEnabled)
        assertTrue(sut.url == "")
    }

    @Test
    fun fromJson_whenInputEmpty_returnsConfigWithDefaultValues() {
        val sut = AnalyticsConfiguration(JSONObject())
        assertFalse(sut.isEnabled)
        assertTrue(sut.url == "")
    }
}
